File: JS/LeetC/arrays/twoSum.js
function twoSum(arr, target) {
	let obj = {};

	for (let i = 0; i < arr.length; i++) {
		if (obj[target - arr[i]] !== undefined) {
			return [i, obj[target - arr[i]]];
		} else {
			obj[arr[i]] = i;
		}
	}
}

console.log(twoSum([2, 7, 11, 15], 9));

File: python/leetcode/arrays/twoSum.py
class Solution:
	def twoSum(self, nums: List[int], target: int) -> List[int]:
		if len(nums) < 2:
			return []
		vals = {}
		for i in range(len(nums)):
			current = nums[i]
			complement = target - current
			if current not in vals:
				vals[current] = i
			if complement in vals and vals[complement] != i:
				return [i, vals[complement]]
